### PR TITLE
Fix custom field endpoints

### PR DESCRIPTION
**Pull request by wuz**
- Add custom field option endpoints
- Bump version
- Fix custom field retrieve and list endpoints

### CODE REVIEW

File: lib/greenhouse_io/api/client.rb [hunk 1]
```diff
@@ -93,12 +93,12 @@ def sources(id = nil, options = {})
       get_from_harvest_api "/sources#{path_id(id)}", options
     end
 
+    def custom_field(id = nil, options = {})
+      get_from_harvest_api "/custom_field#{path_id(id)}", options
+    end
```
Comment (nbcraft) [inline lib/greenhouse_io/api/client.rb:98]: I like this split better 👍

File: lib/greenhouse_io/api/client.rb [hunk 2]
```diff
@@ -93,12 +93,12 @@ def sources(id = nil, options = {})
...
+
     def custom_fields(field_type = nil, options = {})
-      if field_type
-        get_from_harvest_api "/custom_field#{path_id(field_type)}", options
-      else
-        get_from_harvest_api "/custom_fields", options
-      end
+      get_from_harvest_api "/custom_fields#{path_id(field_type)}", options
```
Comment (nbcraft) [inline lib/greenhouse_io/api/client.rb:101]: Almost feel like we should rename `path_id` to `path_param` or something?
```rb
def path_param(param = nil)
  param.present? "/#{param}" : ''
end
```

Comment (nbcraft) [reply lib/greenhouse_io/api/client.rb:101]: Wait is that it?

Calling `client.custom_fields(nil, { page: 12 })` before should have worked then no?
That would have done: `get_from_harvest_api "/custom_fields", { page: 12 }`

What's changed for the pagination here?

Comment (wuz) [reply lib/greenhouse_io/api/client.rb:101]: yeah, so nothing actually changed with pagination. Their API kinda sucks and just always returns the first page if there is only 1 page.

Comment (nbcraft) [reply lib/greenhouse_io/api/client.rb:101]: So... we're still screwed? 🤔

**Review by nbcraft — APPROVED** [review]
This looks fine 👍  but we still have issues with pagination.

Up to you for `path_id` vs. `path_param`

File: lib/greenhouse_io/api/client.rb (hunk 2 above)
Comment (wuz) [reply lib/greenhouse_io/api/client.rb:101]: > Almost feel like we should rename path_id to path_param or something?

Also, as far as renaming, I'm 100% up for it. I am trying to keep things as close to the OG code as possible, but IDK if that is really needed

Comment (nbcraft) [reply lib/greenhouse_io/api/client.rb:101]: Sounds good yeah keep it that way 👍

Comment (wuz) [reply lib/greenhouse_io/api/client.rb:101]: > So... we're still screwed? 🤔

I don't think so, I think there is a way around this, but I gotta investigate a bit more.

**Pull request merged**